service_discovery/dto/service_proposal.go

```diff
@@ -76,7 +128,7 @@ func (genericProposal *ServiceProposal) UnmarshalJSON(data []byte) (err error) {
 		PaymentMethodType string           `json:"payment_method_type"`
 		ServiceDefinition *json.RawMessage `json:"service_definition"`
 		PaymentMethod     *json.RawMessage `json:"payment_method"`
-		ProviderContacts  []Contact        `json:"provider_contacts"`
+		ProviderContacts  *json.RawMessage `json:"provider_contacts"`
```

Comment:
Here You could have type `[]contactJson`, why:
- avoid parsing root json string twice 
- gives You already parsed array of untyped contacts